### ISSUE TITLE
Benches slower than std::Mutex under contention on my mac

### ISSUE THREAD

**Issue by jjl** [stjepang/simple-mutex#1]
Saw this and was curious how much slower it is than parking_lot. I was a bit surprised to find that it benches slower than the std mutex under contention.

```
test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/release/deps/parking_lot-faa69a48c3881858

running 3 tests
test contention    ... bench:     420,674 ns/iter (+/- 73,596)
test create        ... bench:           0 ns/iter (+/- 0)
test no_contention ... bench:     173,168 ns/iter (+/- 10,753)

test result: ok. 0 passed; 0 failed; 0 ignored; 3 measured; 0 filtered out

     Running target/release/deps/simple_mutex-9e73a9d22843fa41

running 3 tests
test contention    ... bench:   1,228,331 ns/iter (+/- 58,846)
test create        ... bench:           0 ns/iter (+/- 0)
test no_contention ... bench:     280,536 ns/iter (+/- 34,175)

test result: ok. 0 passed; 0 failed; 0 ignored; 3 measured; 0 filtered out

     Running target/release/deps/std-9d0590e22b1ad1d8

running 3 tests
test contention    ... bench:   1,051,495 ns/iter (+/- 41,349)
test create        ... bench:         112 ns/iter (+/- 11)
test no_contention ... bench:     336,078 ns/iter (+/- 24,454)

test result: ok. 0 passed; 0 failed; 0 ignored; 3 measured; 0 filtered out
```

**Comment by stjepang** [comment]
Can you try tweaking some numbers? For example, what happens if this loop makes only 5 rather than 10 steps?

https://github.com/stjepang/simple-mutex/blob/9b57c4201ce4ebc90e072349e38fbbbb30bac251/src/lib.rs#L104

**Comment by stjepang** [comment]
Also, I just pushed some changes a published a new version. Make sure to use the latest code.

**Comment by stjepang** [comment]
Published v1.1.5 with even better performance :) Can you run benchmarks again?

**Comment by jjl** [comment]
I happened to notice the notification email for this while looking for something - anything - to distract from this afternoon's horrible task, so absolutely! :)

Congratulations, looks much better

```
     Running target/release/deps/parking_lot-6316358d8373a2d6

running 3 tests
test contention    ... bench:     417,480 ns/iter (+/- 46,357)
test create        ... bench:           0 ns/iter (+/- 0)
test no_contention ... bench:     175,145 ns/iter (+/- 7,333)

test result: ok. 0 passed; 0 failed; 0 ignored; 3 measured; 0 filtered out

     Running target/release/deps/simple_mutex-9095b8e00f3a0125

running 3 tests
test contention    ... bench:     936,209 ns/iter (+/- 48,069)
test create        ... bench:           0 ns/iter (+/- 0)
test no_contention ... bench:     208,379 ns/iter (+/- 15,604)

test result: ok. 0 passed; 0 failed; 0 ignored; 3 measured; 0 filtered out

     Running target/release/deps/std-0dc62b8abf473a82

running 3 tests
test contention    ... bench:   1,056,408 ns/iter (+/- 94,079)
test create        ... bench:         112 ns/iter (+/- 16)
test no_contention ... bench:     333,281 ns/iter (+/- 18,427)

test result: ok. 0 passed; 0 failed; 0 ignored; 3 measured; 0 filtered out
```
